config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/JmsDestinationEntityBuilder.java

```diff
@@ -0,0 +1,129 @@
+                        .map(e -> EntityBuilderHelper.getEntityWithOnlyMapping(CASSANDRA_CONNECTION_TYPE, e.getKey(), idGenerator.generate()))
+                        .collect(Collectors.toList());
+            case ENVIRONMENT:
+                return bundle.getJmsDestinations().entrySet().stream().map(e ->
+                        buildEntity(bundle, e.getKey(), e.getValue(), document)
+                ).collect(Collectors.toList());
+            default:
+                throw new EntityBuilderException("Unknown bundle type: " + bundleType);
+        }
+    }
+
+    private Entity buildEntity(Bundle bundle, String name, JmsDestination jmsDestination, Document document) {
+        String id = idGenerator.generate();
+        
+        // Build JMS Destination element.
+        Element jmsDestinationDetailEle = createElementWithAttributesAndChildren(
+                document,
+                JMS_DESTINATION_DETAIL,
+                ImmutableMap.of(ATTRIBUTE_ID, id),
+                createElementWithTextContent(document, NAME, name),
+                createElementWithTextContent(document, JMS_DESTINATION_NAME, "TODO"),
+                createElementWithTextContent(document, INBOUND, "TODO"),
+                createElementWithTextContent(document, ENABLED, "TODO"),
+                createElementWithTextContent(document, TEMPLATE, jmsDestination.isTemplate()),
+                createElementWithTextContent(document, PROPERTIES, "TODO")
+        );
+        
+        Map<String, Object> destinationDetailsProps = new HashMap<>();
+        // TODO (kpak)
+        destinationDetailsProps.put("username", "");
+        destinationDetailsProps.put("password", "");
+        destinationDetailsProps.put("replyType", "");
+        destinationDetailsProps.put("replyToQueueName", "");
+        destinationDetailsProps.put("useRequestCorrelationId", "");
+        destinationDetailsProps.put("inbound.acknowledgementType", "");
+        destinationDetailsProps.put("inbound.failureQueueName", "");
+        destinationDetailsProps.put("outbound.MessageType", "");
+        destinationDetailsProps.put("inbound.maximumSize", "");
+        buildAndAppendPropertiesElement(destinationDetailsProps, document, jmsDestinationDetailEle);
+
+        // Build JMS Connnection element.
+        String jmsConnectionEleId = idGenerator.generate();
+        Element jmsConnectionEle = createElementWithAttributesAndChildren(
+                document,
+                JMS_CONNECTION,
+                ImmutableMap.of(ATTRIBUTE_ID, jmsConnectionEleId),
+                createElementWithTextContent(document, DESTINATION_TYPE, jmsDestination.getProviderType()),
+                createElementWithTextContent(document, TEMPLATE, jmsDestination.isTemplate())
+        );
+
+        Map<String, Object> connectionProps = new HashMap<>();
+//        connectionProps.put(JNDI_INITIAL_CONTEXT_FACTORY_CLASSNAME, jmsDestination.getInitialContextFactoryClassname());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWc81LiZfimTzgTvo62E&open=AWc81LiZfimTzgTvo62E)

[ISSUE_KEY]: AWc81LiZfimTzgTvo62E